quint/src/effects/base.ts

```diff
@@ -315,27 +347,56 @@ function tryToUnpack(
   const variablesByComponentKind: Map<ComponentKind, Variables[]> = new Map()
 
   // Combine the other effects into a single effect, to be unified with the unpacked effect
-  effects2.forEach(e => {
-    if (e.kind === 'concrete') {
-      e.components.forEach(c => {
-        const variables = variablesByComponentKind.get(c.kind) || []
-        variables.push(c.variables)
-        variablesByComponentKind.set(c.kind, variables)
-      })
-    } else {
-      return left(`Found non concrete effect while trying to unpack: ${effectToString(e)}`)
+
+  // If all the effects are concrete, we compine them into a single concrete
+  // effect by combining the variables of each component of the same kind
+  if (effects2.every(e => e.kind === 'concrete')) {
+    effects2.forEach(e => {
+      if (e.kind === 'concrete') {
+        e.components.forEach(c => {
+          const variables = variablesByComponentKind.get(c.kind) || []
```

Comment:
I did not know that one could use `||` like that, but that's JS/TS. Is not `??` the proper way of doing that in TS?

Reply:
Ah, I can't believe I still have ruby habits after more than 1 year. Weird that the typechecker didn't say anything, I don't think the semantics are the same. I'll fix it, good catch